src/pow.rs

```diff
@@ -0,0 +1,1164 @@
+// Rust Bitcoin Library - Written by the rust-bitcoin developers.
+// SPDX-License-Identifier: CC0-1.0
+
+//! Proof-of-work related integer types.
+//!
+//! The functions here are designed to be fast.
+//!
+
+macro_rules! impl_from_and_impl {
+    ($ty:ident) => {
+        impl From<[u64; 4]> for $ty {
```

Comment:
Implicit endianess is not great, I'd prefer to remove it and have a methods similar to `from_be_bytes`, `to_be_bytes` and their `le` counterparts.

Reply:
Added `from_be`, `from_le`, `to_be`, `to_le`.